patches/chromium/fix_hunspell_crash.patch

```diff
@@ -0,0 +1,147 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Cheng Zhao <zcbenz@gmail.com>
+Date: Thu, 4 Oct 2018 14:57:02 -0700
+Subject: Make sure hunspell file is not destroyed in UI thread
+
+Submitted to Chromium at:
+https://chromium-review.googlesource.com/c/chromium/src/+/2206199/1
+
+diff --git a/chrome/browser/spellchecker/spellcheck_hunspell_dictionary.cc b/chrome/browser/spellchecker/spellcheck_hunspell_dictionary.cc
+index 653ef44ff095015ef25cc59bf42f9fbd9d907160..d6f9bf8e0d2cf2e52698d4d132361a5d7118d00c 100644
+--- a/chrome/browser/spellchecker/spellcheck_hunspell_dictionary.cc
++++ b/chrome/browser/spellchecker/spellcheck_hunspell_dictionary.cc
+@@ -93,21 +93,28 @@ bool SaveDictionaryData(std::unique_ptr<std::string> data,
+ 
+ }  // namespace
+ 
+-SpellcheckHunspellDictionary::DictionaryFile::DictionaryFile() {
+-}
++SpellcheckHunspellDictionary::DictionaryFile::DictionaryFile(
++    base::TaskRunner* task_runner) : task_runner_(task_runner) {}
```

Comment:
Shouldn't `DictionaryFile` maintain the task runner as a `refptr` ?

If `SpellcheckHunspellDictionary` is destroyed before `InitializeDictionaryLocation` as mentioned in the CL, there is no guarantee for the task runner to be alive till the `DictionaryFile` is destroyed.

Reply:
The `task_runner_` is stored as `scoped_refptr` in `DictionaryFile`, it is just passed as raw ptr.

Reply:
Missed the declaration, what is the advantage of passing as raw ptr ?

Reply:
Passing raw ptr could save the atomic Add/Release Refs when copying scoped_ptr, but it was before the `std::move`.

After checking https://chromium.googlesource.com/chromium/src/+/lkgr/styleguide/c++/c++.md#object-ownership-and-calling-conventions, it seems that now Chromium recommends passing scoped_ptr.

Reply:
Yeah being explicit on the ownership is better. It would be good if you can change in the upstream CL, otherwise we can leave this as such. Thanks for the clarification!